transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java

```diff
@@ -282,14 +282,29 @@ public BackendResponse send(BackendRequest request) {
     // CCT backend supports 2 different endpoints
     // We route to CCT backend if extras are null and to LegacyFlg otherwise.
     // This (anti-) pattern should not be required for other backends
-    final String apiKey =
-        request.getExtras() == null ? null : LegacyFlgDestination.decodeExtras(request.getExtras());
+    URL actualEndPoint = endPoint;
+    String apiKey = null;
+    if (request.getExtras() != null) {
+      try {
+
+        LegacyFlgDestination destination =
+            LegacyFlgDestination.parseFromExtras(request.getExtras());
+        apiKey = destination.getAPIKey();
+        String destinationEndPoint = destination.getEndPoint();
+        if (destinationEndPoint != null && !destinationEndPoint.isEmpty()) {
+          actualEndPoint = parseUrlOrThrow(destination.getEndPoint());
+        }
+      } catch (IllegalStateException e) {
+        Logging.e(LOG_TAG, "Couldn't parse extras", e);
+        return null;
```

Comment:
I don't think we can return null from this method. I think it's appropriate to return `BackendResponse.fatal()` instead.